compiler/hash-target/src/abi.rs

```diff
@@ -83,6 +83,22 @@ impl Integer {
         }
     }
 
+    /// Finds the smallest [Integer] type which can represent the signed
+    /// value.
+    #[inline]
+    #[allow(clippy::match_overlapping_arm)]
+    pub fn fit_signed(value: i128) -> Integer {
+        use Integer::*;
+
+        match value {
```

Comment:
Why is clippy saying overlapping arm?

Reply:
Since each succeeding arm overlaps the previous one, it complains that you shouldn't do this.

Reply:
Right makes sense